azkaban-common/src/main/java/azkaban/utils/YarnUtils.java

```diff
@@ -107,17 +112,32 @@ public static List<ApplicationReport> getAllAliveAppReportsByExecIDs(final YarnC
 
 
   /**
-   * Uses YarnClient to kill the jobs one by one
+   * Uses YarnClient to kill the jobs one by one, each kill has a timeout
    */
   public static void killAllAppsOnCluster(YarnClient yarnClient, Set<String> applicationIDs,
       Logger log) {
     log.info(String.format("Killing applications: %s", applicationIDs));
+    ExecutorService executor = Executors.newSingleThreadExecutor();
 
     for (final String appId : applicationIDs) {
+      Future<?> future = executor.submit(new Runnable() {
```

Comment:
Has this been tested?

Reply:
let me add some unit tests

Reply:
Do you mind put the execution link here? and any log line saying about the timeout?

Reply:
I posted in our internal channel

Reply:
13-12-2022 11:46:54 PST pigWordCount1 WARN - Timed out killing the yarn app application_1670314587930_1089113, cancelling the runnable...